servers/rendering/renderer_rd/forward_mobile/render_forward_mobile.cpp

```diff
@@ -481,9 +483,8 @@ RID RenderForwardMobile::_setup_render_pass_uniform_set(RenderListType p_render_
 
 	/* we have limited ability to keep textures like this so we're moving this to a set we change before drawing geometry and just pushing the needed texture in */
 	{
-		RD::Uniform u;
-		u.binding = 6;
-		u.uniform_type = RD::UNIFORM_TYPE_TEXTURE;
+		Vector<RID> textures;
+		textures.resize(scene_state.max_lightmaps * 2);
```

Comment:
If this function is called a lot, you probably want to use a stack fixed array. Basically a fancy fixed C array with all the usual machinery for `push_back` etc.

I have a basic one I wrote for 3.x (`core/fixed_array.h`) but you can equally well modify `LocalVector` template to be capable of storing on the stack (let me know if this is of interest, I recently did this for a third party module).

Reply:
Ah I see it's getting pass to RD::Uniform below and stored there in which case maybe this allocation is unavoidable. :slightly_frowning_face:

Reply:
I think the better option here is just to track the textures that may have changed and only recreate the Uniforms array when we know there will be a change. 

Basically this function recreates the array of Uniforms every frame and then indexes into a hash map to see if we have cached this uniform set or not. Instead, with minimal tracking, we can just check if any Uniform actually changed, then only run this code when it has. Realistically, this won't run most frames. 

But that is a riskier change to make, so I'd like to do it in a follow up PR as this gives us 99% of the benefit and is totally safe